### ISSUE TITLE
Ability to open files in search without changing focus

### ISSUE THREAD

**Issue by pfrcks** [nvim-pack/nvim-spectre#117]
Currently <cr> opens the file under search results and moves the focus to the opened file. Instead I was wondering if it would be possible to open the file without changing the focus from spectre results window?

Use case: sometimes I want to see the context of the search phrase. Changing focus on <cr> introduces a lot of friction where I have to traverse back to the spectre window to move to the next result.

**Comment by EpsilonKu** [comment]
Working on it. Lmao.

**Comment by EpsilonKu** [comment]
Yeah, I already windwp to remove it. Currently working on other issue.

**Comment by windwp** [comment]
you can add pinned tag to avoid bot :smiling_face_with_tear:

**Comment by EpsilonKu** [comment]
@pfrcks can u try #159 with `require('spectre.actions').open_entry()`. If it works find I will add to default keybindings.

**Comment by matiboy** [comment]
Quite late but I was looking for the same and @EpsilonKu `open_entry` isn't exposed. The below does the trick for me, basically open and refocus on Spectre window:

```lua
-- <Tab>: open file in next window but stay in Spectre
vim.keymap.set("n", "<Tab>", function()
  local spectre_win = vim.api.nvim_get_current_win()
  require("spectre.actions").select_entry()
  if vim.api.nvim_win_is_valid(spectre_win) then
    vim.api.nvim_set_current_win(spectre_win)
  else
    vim.notify("Invalid window ID: " .. spectre_win, vim.log.levels.WARN)
  end
end, { buffer = true, desc = "Open entry, stay in Spectre" })
```

I had to wrap the override setting of keymaps into a `vim.schedule(function()` for some reason.

I personally also like to open a file and close the search which I did the following way if anyone is interested:

```lua
-- <CR> open result and close
vim.keymap.set("n", "<CR>", function()
    local spectre_win = vim.api.nvim_get_current_win()
	require("spectre.actions").select_entry()
    vim.api.nvim_win_close(spectre_win, false)
end, { buffer = true, desc = "Open file and close Spectre" })
```

**Comment by EpsilonKu** [comment]
alr, I will try to open new pr soon
